### PR TITLE
Feature/merge signable hashable

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
Merge Hashable and Signable class of shared_model into one Signable. Actually we don't have hashable but not signable in the model.
Some classes (Asset and so on) were inherited form Hashable to perform operator==. Now they are reworked with their own comparison operators.

### Benefits
Simplify inheritance

### Possible Drawbacks 
no

### Usage Examples or Tests *[optional]*
no

### Alternate Designs *[optional]*
no

### CODE REVIEW

File: shared_model/interfaces/base/signable.hpp [hunk 1]
```diff
@@ -117,6 +113,9 @@ namespace shared_model {
                        [](auto &signature) { return signature->toString(); })
             .finalize();
       }
+
+     protected:
+      mutable boost::optional<types::HashType> hash_;
```
Comment (luckychess) [inline shared_model/interfaces/base/signable.hpp]: Why not private?

File: shared_model/interfaces/base/signable.hpp [hunk 2]
```diff
@@ -79,33 +80,28 @@ namespace shared_model {
...
-              Hashable<Model, OldModel>::HashProviderType::makeHash(payload()));
-        }
-        return *Hashable<Model, OldModel>::hash_;
-      }
-#else
-      const typename types::HashType &hash() const override {
-        if (Hashable<Model>::hash_ == boost::none) {
-          Hashable<Model>::hash_.emplace(
-              Hashable<Model>::HashProviderType::makeHash(payload()));
+      virtual const types::HashType &hash() const {
```
[the remark this answers is not in the document]
Comment (Alexey-N-Chernyshov) [reply shared_model/interfaces/base/signable.hpp]: solved

[the remark this answers is not in the document]
Comment (l4l) [reply shared_model/interfaces/base/signable.hpp]: how?

Comment (l4l) [inline shared_model/interfaces/base/signable.hpp]: Why virtual?

[the remark this answers is not in the document]
Comment (Alexey-N-Chernyshov) [reply shared_model/interfaces/base/signable.hpp]: It was used to access base class Hashable<>::hash. Now classes are merged and hash moved to Signable, so there is no need in template.

**Pull request merged**